### ISSUE TITLE
Improved error message for multiple expressions

### ISSUE THREAD

**Issue by braaannigan** [pola-rs/polars#7395]
### Problem description

The following now generates an error message that I think could be improved:
```python
import polars as pl
df = pl.DataFrame({'lag1':[0,1,None],'lag2':[0,None,2]})
df.filter(pl.col('^lag.*$').is_not_null())

ComputeError: The predicate passed to 'LazyFrame.filter' expanded to multiple expressions:

	col("lag1").is_not_null().any(),
	col("lag2").is_not_null().any(),
This is ambiguous. Try to combine the predicates with the 'all' or `any' expression.
```
As @ghuls pointed out on discord this can be resolved with
```python
df.filter(pl.all(pl.col('^lag.*$').is_not_null()))
```
I think the error message needs to be more explicit as it wasn't obvious to me how to solve this. It may be easiest to add a link to an example in the python API in which case I could take it on

**Comment by ritchie46** [comment]
What is your proposal then? I tried to make it clear by showing which expressions it expands to and give a hint to combine the predicates with `all` or `any`.

If we want to explicitly show the original expression we have to clone it first, this would add some latency to the `filter` expression which I am not sure we need to do because of our users who need very low latency answers.

**Comment by braaannigan** [comment]
I'm proposing that we:
- add a docstring example in filter
- in the error message a link to the filter API page with that example

**Comment by ritchie46** [comment]
I don't think we should link to our API docs in code. That can age pretty badly. If do something like that we should make an error registry, but currently that is too much work IMO.

I do think we can add a docstring example. :+1:

**Comment by ghuls** [comment]
Docstring examples in fliter make the most sense to me.

**Comment by SydneyUni-Jim** [comment]
You now need to use `pl.all_horizontal` or `pl.any_horizontal`.

```python
import polars.selectors as cs
df.filter(
  pl.all_horizontal(
   cs.starts_with('lag').is_not_null()
  )
)
```

**Comment by coastalwhite** [comment]
Closing for now as this seems to be resolved.
